Apps/Src/ReadCarCAN.c

```diff
@@ -67,98 +93,145 @@ static inline void chargingEnable(void) {
     // if nothing is precharging and we need to turn the array on, turn it on
     shouldRestartArray = !restartingArray && (Contactors_Get(ARRAY_CONTACTOR)==OFF) && (Contactors_Get(ARRAY_PRECHARGE)==OFF);
 
-    // start precharge for array 
+    // wait for precharge for array 
     if(shouldRestartArray){
 
-         if (Contactors_Set(ARRAY_PRECHARGE, ON) == SUCCESS) {    // only run through precharge sequence if we successfully start precharge
-            restartingArray = true;
+            // Wait to make sure precharge is finished and then restart array
```

Comment:
`restartingArray` is never actually set to `true` anywhere in the code. I think this is where it should be set, since this is the point where we're sure we're restarting the array.